lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -391,7 +391,8 @@ void SPIRVRegularizeLLVMBase::expandSYCLHalfUsing(Module *M) {
     if (F.getName().startswith("_Z28__spirv_VectorExtractDynamic") &&
         F.hasStructRetAttr()) {
       auto *SRetPtrTy = cast<PointerType>(F.getArg(0)->getType());
-      if (isSYCLHalfType(SRetPtrTy->getPointerElementType()))
+      if (isSYCLHalfType(SRetPtrTy->getPointerElementType()) ||
+          isSYCLBfloat16Type(SRetPtrTy->getPointerElementType()))
         ToExpandVEDWithSYCLHalfSRetArg.push_back(&F);
```

Comment:
I think we should rename this container and the others to something like "ToExpandVEDWithSYCLTypeSRetArg" and also rename the corresponding functions, since now there is not only `half` type.

Reply:
Changed it in [96a0d79](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1427/commits/96a0d798373e50b1a7402f233d1bc616c9a87a46)